otri/database/database_adapter.py

```diff
@@ -1,31 +1,193 @@
-from .database_query import DatabaseQuery
-from .database_data import DatabaseData
+"""
+Basic SQLAlchemy wrapper providing some common functionalities for interfacing with a database.
+"""
+
+from typing import List, Mapping, Union
+from contextlib import contextmanager
+from sqlalchemy import *
+from sqlalchemy.orm import sessionmaker
+from sqlalchemy.ext.automap import automap_base
+
+from ..utils import logger as log
 
 
-'''
-Classes:
-DatabaseAdapter -- Abstract class
-'''
 class DatabaseAdapter:
     '''
     Abstract class used to access with the same methods independently from the kind of database used.
+    Uses SQLAlchemy to interface with the database.
     '''
-    def write(self, data : DatabaseData):
+
+    def __init__(self, password: str, host: str, port: Union[str, int], user: str, database: str):
+        '''
+        Parameters:
+            host : str\n
+                The database host address.\n
+            port : Union[str, int]\n
+                The port on which the database is listening.\n
+            user : str\n
+                The username.\n
+            password : str\n
+                The password for the given user.\n
+            database : str\n
+                The database on which to connect.
+        '''
+        try:
+            log.i("Trying to connect to database.")
```

Comment:
Upper cases on logs look bad.
Wanted to edit it somewhere else but didn't want to pollute other PRs, you could do it in this one.

Reply:
U sure about that? I don't dislike them

Reply:
Idk I'm a bit confused too, I don't like an upper case in the middle of a log but it could make sense.
I've got my own demons to fight.

Let's leave that could be either, doesn't change much.